ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -200,51 +222,53 @@ splitAfterTxSize (TxSeq txs) n =
 --
 -- This function is used to verify whether 'splitAfterTxSize' behaves as
 -- expected.
-splitAfterTxSizeSpec :: TxSeq tx -> TxSizeInBytes -> (TxSeq tx, TxSeq tx)
+splitAfterTxSizeSpec :: forall sz tx.
+     Measure sz
+  => TxSeq sz tx
+  -> sz
+  -> (TxSeq sz tx, TxSeq sz tx)
 splitAfterTxSizeSpec txseq n =
-    mapTuple fromList $ go 0 [] (toList txseq)
+    mapTuple fromList $ go Measure.zero [] (toList txseq)
   where
     mapTuple :: (a -> b) -> (a, a) -> (b, b)
     mapTuple f (x, y) = (f x, f y)
 
-    go :: TxSizeInBytes
-       -> [TxTicket tx]
-       -> [TxTicket tx]
-       -> ([TxTicket tx], [TxTicket tx])
+    go :: sz
+       -> [TxTicket sz tx]
+       -> [TxTicket sz tx]
+       -> ([TxTicket sz tx], [TxTicket sz tx])
     go accByteSize accTickets = \case
       []
         -> (reverse accTickets, [])
       t:ts
-        | let accByteSize' = accByteSize + txTicketTxSizeInBytes t
-        , accByteSize' <= n
+        | let accByteSize' = accByteSize `Measure.plus` txTicketSize t
```

Comment:
The `Byte` in `accByteSize` seems outdated. Also, `go` could be removed by using [`Measure.splitAt`](https://intersectmbo.github.io/cardano-base/measures/Data-Measure.html#v:splitAt).

Reply:
I fixed the name, but left the "specification" function as transparent as possible 👍